apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V97__Synthea_Pre_Validation.sql

```diff
@@ -0,0 +1,324 @@
+                            );
+                        END IF;
+
+                        IF RSLT < 1 THEN
+${logic.psql-only}          RSLT :=
+${logic.hsql-only}          set RSLT =
+                            (
+                                select count(clm_id) from snf_claims
+                                where clm_grp_id <= p_beg_clm_grp_id and clm_grp_id > CLM_GRP_ID_END
+                                limit 1
+                            );
+                        END IF;
+
+                        IF RSLT < 1 THEN
+${logic.psql-only}          RSLT :=
+${logic.hsql-only}          set RSLT =
+                            (
+                                select count(clm_id) from dme_claims
+                                where clm_grp_id <= p_beg_clm_grp_id and clm_grp_id > CLM_GRP_ID_END
+                                limit 1
+                            );
+                        END IF;
+
+                        IF RSLT < 1 THEN
+${logic.psql-only}          RSLT :=
+${logic.hsql-only}          set RSLT =
+                            (
+                                select count(clm_id) from hha_claims
+                                where clm_grp_id <= p_beg_clm_grp_id and clm_grp_id > CLM_GRP_ID_END
+                                limit 1
+                            );
+                        END IF;
+
+                        IF RSLT < 1 THEN
+${logic.psql-only}          RSLT :=
+${logic.hsql-only}          set RSLT =
+                            (
+                                select count(clm_id) from hospice_claims
+                                where clm_grp_id <= p_beg_clm_grp_id and clm_grp_id > CLM_GRP_ID_END
+                                limit 1
+                            );
+                        END IF;
+
+                        IF RSLT < 1 THEN
+${logic.psql-only}          RSLT :=
+${logic.hsql-only}          set RSLT =
+                            (
+                                select pde_id from partd_events
+                                where pde_id <= p_pde_id_start
+                                and clm_grp_id <= p_beg_clm_grp_id and clm_grp_id > CLM_GRP_ID_END
+                                limit 1
+                            );
+                        END IF;
+
+    -- look for hicn or mbi collisions...only need 1 collision to trigger a problem
+                        IF RSLT < 1 THEN
+${logic.psql-only}          RSLT :=
+${logic.hsql-only}          set RSLT =
+                            (
+                                select count (bene_id) from beneficiaries
+                                where (hicn_unhashed = p_hicn_start or mbi_num = p_mbi_start)
+                                limit 1    
+                            );
+                        END IF;
+
+    -- add support for fi_cntl_num here; not the most efficient step since
+    -- fi_doc_clm_cntl_num is not an indexed field (therefore causing a
+    -- full table scan).
```

Comment:
Is this a TODO?

Reply:
removed; artifact from the original query which was a giant 'union' which didn't work so well in HSQL.